cmd/apiserver/app/server/run_server.go

```diff
@@ -42,7 +42,8 @@ func RunServer(opts *ServiceCatalogServerOptions) error {
 func runTPRServer(opts *ServiceCatalogServerOptions) error {
 	tprOpts := opts.TPROptions
 	glog.Infoln("Installing TPR types to the cluster")
-	if err := tpr.InstallTypes(tprOpts.clIface); err != nil {
+	tprInstaller := tpr.NewInstallerTPRs(tprOpts.clIface)
```

Comment:
Regarding the code for the fakes. I am not sure the code for the mocks would complicate the project in anyway. We could generally ignore all `fakes` directories.
While it is true we can make the code footprint smaller by hand-coding the mocks, in my opinion it is largely inefficient to do so since we can easily generate mocks from it's interface, and we have plenty of proven tools for it. The complexity of hand-coding also depends on the interface size.
Lastly, hand-coding requires code changing every time the interface changes (for example new `func()`), but with generated code, we can avoid that by auto `go generate`